src/ifcopenshell-python/ifcopenshell/util/mvd_info.py

```diff
@@ -0,0 +1,310 @@
+        self._update_keyword("Comment", value)
+
+    @property
+    def exchange_requirements(self):
+        self._ensure_parsed()
+        return self._parsed.exchange_requirements if self._parsed else None
+
+    @exchange_requirements.setter
+    def exchange_requirements(self, new_value: str):
+        self._update_keyword("ExchangeRequirement", new_value)
+
+    @property
+    def options(self):
+        self._ensure_parsed()
+        if isinstance(self._parsed.options, dict):
+            return DictionaryHandler(self._parsed.options, self, "Option")
+        return self._parsed.options if self._parsed else None
+
+    @options.setter
+    def options(self, new_value: str):
+        self._update_keyword("Option", new_value)
+
+    @property
+    def keywords(self):
+        self._ensure_parsed()
+        return self._parsed.keywords if self._parsed else set()
+
+    def _update_keyword(self, keyword: str, new_value: str):
+        updated = False
+        new_line = f"{keyword} [{new_value}]"
+        lines = []
+        for line in self.description:
+            if line.strip().startswith(f"{keyword} ["):
+                lines.append(new_line)
+                updated = True
+            else:
+                lines.append(line)
+        if not updated:
+            lines.append(new_line)
+        self.description = lines
+
+    def __getattr__(self, name):
+        self._ensure_parsed()
+        if hasattr(self._parsed, '_dynamic'):
+            name_lc = name.lower()
+            if name_lc in self._parsed._dynamic:
+                value, original_keyword = self._parsed._dynamic[name_lc]
+                return DictionaryHandler(value, self, original_keyword)
+        raise AttributeError(f"'MvdInfo' object has no attribute '{name}'")
+
+    def __dir__(self):
+        base = super().__dir__()
+        if self._parsed and hasattr(self._parsed, '_dynamic'):
+            return base + [kw for _, kw in self._parsed._dynamic.values()]
+        return base
+
+
+class DictionaryHandler(dict):
+    def __init__(self, initial_data, mvdinfo, keyword):
+        super().__init__()
+        self._mvdinfo = mvdinfo
+        self._keyword = keyword
+        for k, v in initial_data.items():
+            if isinstance(v, list):
+                super().__setitem__(k, AutoCommitList(v, self._commit))
+            else:
+                super().__setitem__(k, v)
+
+    def _commit(self):
+        new_value = "; ".join(
+            f"{k}: {', '.join(v) if isinstance(v, list) else v}"
+            for k, v in self.items()
+        )
+        self._mvdinfo._update_keyword(self._keyword, new_value)
+
+    def __setitem__(self, key, value):
+        if isinstance(value, list):
+            value = AutoCommitList(value, self._commit)
+        super().__setitem__(key, value)
+        self._commit()
+
+    def __delitem__(self, key):
+        super().__delitem__(key)
+        self._commit()
+
+
+class AutoCommitList(list):
```

Comment:
ok, this is maybe a bit more involved than I had in mind. 

A simple `to_string()` for example on MvdInfo would have been ok, so that you would have to manually call that and assign back to the header if you're mutating your MvdInfo. But I don't see downsides to your approach, so let's roll with this and see how it goes :)

Reply:
I thought about something like that, but also considered that it would be more intuitive if we could directly modify it through this mvd parser. While wondering whether it would be possible I started implementing it ..  
Maybe it is more error-prone and/or not as intuitive as expected. We can then always reverse it. Another option would be to have both methods available. 

I've also made sure we can add view_definitions this way, e.g.
`f.mvd.view_definitions.append('AlignmentBased-View')`